exist-core/src/main/java/org/exist/backup/restore/RestoreHandler.java

```diff
@@ -303,17 +303,12 @@ private DeferredPermission restoreResourceEntry(final Attributes atts) throws SA
             }
         }
 
-        MimeType mimeType = null;
-        if (xmlType) {
-            mimeType = MimeType.XML_TYPE;
-        } else if (mimeTypeStr != null) {
-            mimeType = MimeTable.getInstance().getContentType(mimeTypeStr);
-            if (mimeType == null) {
-                mimeType = MimeType.BINARY_TYPE;
-            } else if (mimeType.isXMLType() !=  xmlType) {
-                listener.warn("Calculated mime type " + mimeType + " type differs from the one specified in the backup, adjusting it");
-                mimeType = new MimeType(mimeType.getName(), xmlType ? MimeType.XML : MimeType.BINARY);
-            }
+        final MimeType mimeType;
+        if (mimeTypeStr == null || mimeTypeStr.trim().isEmpty()) {
+            mimeType = xmlType ? MimeType.XML_TYPE : MimeType.BINARY_TYPE;
+            listener.warn("Missing mimetype attribute in the backup __contents__.xml file for: " + name + ", assuming: " + mimeType);
+        } else {
+            mimeType = new MimeType(mimeTypeStr.trim(), xmlType ? MimeType.XML : MimeType.BINARY);
```

Comment:
Could this be moved into a protected method on a new parent class to be used on this and the `SystemImportHandler` in order to have no code duplication? In a later step there are some other duplicate methods, that could be pulled up in a second step later...

Reply:
@reinhapa I think that sort of refactoring is more than I want to achieve in this PR. So I have opened a new PR here - https://github.com/eXist-db/exist/pull/4392 that takes care of the refactoring and could be merged first if you like? I could then rebase this atop it.